colossalai/kernel/cuda_native/setup.py

```diff
@@ -0,0 +1,252 @@
+                    "csrc/attention_infer_kernels/flash_attn/src/flash_fwd_hdim224_fp16_sm80.cu",
+                    "csrc/attention_infer_kernels/flash_attn/src/flash_fwd_hdim224_bf16_sm80.cu",
+                    "csrc/attention_infer_kernels/flash_attn/src/flash_fwd_hdim256_fp16_sm80.cu",
+                    "csrc/attention_infer_kernels/flash_attn/src/flash_fwd_hdim256_bf16_sm80.cu",
+                ],
+            extra_compile_args={
+                "cxx": ["-O3", "-std=c++17"] + generator_flag,
+                "nvcc": append_nvcc_threads(
+                    [
+                        "-O3",
+                        "-std=c++17",
+                        "-U__CUDA_NO_HALF_OPERATORS__",
+                        "-U__CUDA_NO_HALF_CONVERSIONS__",
+                        "-U__CUDA_NO_HALF2_OPERATORS__",
+                        "-U__CUDA_NO_BFLOAT16_CONVERSIONS__",
+                        "--expt-relaxed-constexpr",
+                        "--expt-extended-lambda",
+                        "--use_fast_math",
+                        "--ptxas-options=-v",
+                        "-lineinfo"
+                    ]
+                    + generator_flag
+                    + cc_flag
+                ),
+            },
+            include_dirs=[
+                Path(this_dir) / 'csrc'/'attention_infer_kernels'/'flash_attn' ,
+                Path(this_dir) / 'csrc'/ 'attention_infer_kernels'/'flash_attn' / 'src',
+                Path(this_dir)  / 'csrc'/'cutlass' / 'include',
+            ],
+        ),
+
+        CUDAExtension(
+                name="col_linear_lib",
+                sources=[
+                    "csrc/attention_infer_kernels/linear/linear_op.cpp",
+                    "csrc/attention_infer_kernels/linear/gemm.cu",
+                ],
+            extra_compile_args={
+                "cxx": ["-O3", "-std=c++17"] + generator_flag,
+                "nvcc": append_nvcc_threads(
+                    [
+                        "-O3",
+                        "-std=c++17",
+                        "-U__CUDA_NO_HALF_OPERATORS__",
+                        "-U__CUDA_NO_HALF_CONVERSIONS__",
+                        "-U__CUDA_NO_HALF2_OPERATORS__",
+                        "-U__CUDA_NO_BFLOAT16_CONVERSIONS__",
+                        "--expt-relaxed-constexpr",
+                        "--expt-extended-lambda",
+                        "--use_fast_math",
+                        "--ptxas-options=-v",
+                        "-lineinfo"
+                        "-lcublas"
+                    ]
+                    + generator_flag
+                    + cc_flag
+                ),
+            },
+            include_dirs=[
+                Path(this_dir) / 'csrc'/'attention_infer_kernels' /'linear' ,
+                Path(this_dir)  / 'csrc'/'cutlass' / 'include',
+            ],
+        ),
+
+    ],
+    cmdclass={
+        'build_ext': BuildExtension
+})
+
+
```

Comment:
Can you integrate this to colossalai's setup?

Reply:
yes!